app/models/article.rb

```diff
@@ -21,6 +22,8 @@ class Article < ApplicationRecord
   before_validation :generate_published_dates, on: [:create, :update]
   before_validation :downcase_content_format,  on: [:create, :update]
 
+  after_save :create_redirect
```

Comment:
I'm guessing the `Redirect` being created is mandatory, I would think doing a `before_save` would be better. If the `Redirect` fails to save with the `after_save`, the `Article` has already saved with a short path which doesn't have a redirect.